### DISCUSSION TITLE
Assignment for the portfolio

### DISCUSSION THREAD

**Discussion by ellenalamont17** [earthlab-education/Earth-Analytics-2023-01-Intro#203] [Q&A02 Open Reproducible Science]
Could we please get clarification on how to bring our second temperature analysis over to our personal portfolios? I ended up making all of the changes in this week's classroom codespace, but I am not sure how to set it up in my personal repository.

**Comment by advythr** [comment]
As a follow-up question to this, when we redo the analysis for the second city, are we supposed to write the temperature analysis in the classroom codespace or create a new Python notebook file for it?

**Reply by eculler** [reply]
Hi @advythr, either option is fine. Just note that to use a codespace in your portfolio repository you will need to copy over a file called .devcontainer.json to your portfolio repo. I will have a video for you tomorrow! (And note that the portfolio post is not due until Friday so we can cover this in section as well)

**Comment by eculler** [comment]
@ellenalamont17 See my response to @advythr re: video. But if you're working on this before then, I'd recommend using your portfolio repository. If you've already started in your assignment repository, you can download the notebook file (and the .devcontainer.json file while you're at it) and upload them to the other repository. You can download files either from VSCode (Codespaces) by right-clicking on the file, or by opening the file in GitHub and clicking the download button on the upper right.

**Comment by edchanonline** [comment]
There is an option to Export (top right if I remember correctly). I ran into an issue where saving locally didn't work so I ended up generating an HTML file in the assignment repository then I copied the HTML code into a file I created on my local machine. Upon opening that HTML file, I saved the image as a JPG. Having the JPG, I uploaded it to my portfolio repository. I do wonder if we are supposed to save a copy of the Jupyter Notebook and upload that into our portfolio repo (but then everyone would have my answers!)

**Reply by edchanonline** [reply]
Pictures for illustration:

<img width="1189" alt="Screenshot 2023-09-12 at 3 46 05 PM" src="https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/4795392/ec5c60e4-2125-4dd3-9d81-bb545e54bbfc">
<img width="656" alt="Screenshot 2023-09-12 at 3 46 14 PM" src="https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/4795392/4c64864e-fc03-4e5d-b316-62e4f52783a3">
<img width="677" alt="Screenshot 2023-09-12 at 3 46 21 PM" src="https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/4795392/99196958-38ff-4433-8d42-34eb9448fa6a">

**Reply by edchanonline** [reply]
And there is a download button next to the graph!
<img width="262" alt="Screenshot 2023-09-12 at 3 52 36 PM" src="https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/4795392/e54916e9-52b0-448a-9936-4f26b3c22718">

**Reply by edchanonline** [reply]
Disregard the instructions above - though it's a valid way.

Since I have posted this, there was a new demo video by Elsa at https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/discussions/214 suggesting you can create a devcontainer with the notebooks in your portfolio repo. 

1. Create `.devcontainer.json`
2. Open the codespace
3. Create `notebooks` folder
4. Create your analysis notebook (the `ipynb` file)
5. Add the following to the beginning of the notebook
```
import warnings

warnings.filterwarnings('ignore')
```
7. Add the following to the end of the notebook
```
%%capture
%%bash
jupyter nbconvert <your-notebook-name>.ipynb --to html --no-input
```
8. Run the notebook. Step 7 will generate a HTML file
9. Commit and sync in your codespace
10. You should be able to open that HTML file like this one at https://eggvoice.github.io/notebooks/ncei_tavg_san_francisco.html

Similar instructions at our wiki at https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/wiki/How-to...#post-your-analysis-to-your-portfolio
